Merges blocked, possible type checking error in TypeScript SDK

An integration test jobs is failing with errors on running `tsc`, preventing PRs from merging.

See: https://github.com/pulumi/pulumi/actions/runs/3474952430/jobs/5808746910#step:36:509

```
=== FAIL: integration TestESMTSCompiled
...
$ tsc
node_modules/@types/node/globals.d.ts:72:13 - error TS2403: Subsequent variable declarations must have the same type.  Variable 'AbortSignal' must be of type '{ new (): AbortSignal; prototype: AbortSignal; abort(reason?: any): AbortSignal; timeout(milliseconds: number): AbortSignal; }', but here has type '{ new (): AbortSignal; prototype: AbortSignal; }'.

72 declare var AbortSignal: {
               ~~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:2071:13
    2071 declare var AbortSignal: {
                     ~~~~~~~~~~~
    'AbortSignal' was also declared here.


Found 1 error in node_modules/@types/node/globals.d.ts:72

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

    command.go:103: Wrote output to /tmp/p-it-fv-az207-6-esm-ts-com-8672e73e-2561626402/command-output/yarn-build.20221115-233835.a0f5b.log
    command.go:34: **** Invoke '/opt/hostedtoolcache/node/14.20.1/x64/bin/yarn run build' in '/tmp/p-it-fv-az207-6-esm-ts-com-8672e73e-2561626402'
    command.go:83: Invoke '/opt/hostedtoolcache/node/14.20.1/x64/bin/yarn run build' failed: exit status 2
yarn run v1.22.19
```

--- a/tests/integration/nodejs/esm-ts-compiled/package.json
+++ b/tests/integration/nodejs/esm-ts-compiled/package.json
@@ -7,7 +7,7 @@
         "build": "tsc"
     },
     "devDependencies": {
-        "@types/node": "^17.0.5"
+        "@types/node": "^18.0.0"
     },
     "peerDependencies": {
         "@pulumi/pulumi": "latest"

